### PR TITLE
Modernize logging and remove all usage of Boost

### PR DESCRIPTION
**Pull request by argilo**
GNU Radio's logging framework now uses fmtlib instead of Boost.Format. GNU radio is also moving away from logging macros. I've updated logging here to match GNU Radio conventions.

I also switched a couple other instances of boost::format to standard C++.

Signed-off-by: Clayton Smith <argilo@gmail.com>

### CODE REVIEW

**Comment by argilo** [comment]
Note: I did not do any testing, I only verified that this builds against GNU Radio's `maint-3.10` and `main` branches.

**Comment by daniestevez** [comment]
Great! I was thinking of doing this, but the thing holding me back was that I didn't know if this "modernized logging" was available in GNU Radio 3.9. Currently the main branch of gr-satellites targets both GNU Radio 3.9 and 3.10, and releases on that branch should work with both versions (Although a maint-3.9 branch could be created if there is enough need, but having a single branch for 3.9 and 3.10 simplifies maintenance).

The pipeline for 3.9 has passed, and I've seen this uses 3.9.5. So it seems that the modern logging also works with 3.9. Do you know if it also works with older 3.9 releases? I'll test with the 3.9.4 I have on my machine.

**Review by daniestevez — APPROVED** [review]

**Comment by argilo** [comment]
As far as I know, the transition to fmtlib only happened in 3.10, so I'd be surprised if this works properly in 3.9. I'll try building locally to see how it goes.

**Comment by daniestevez** [comment]
I'm testing it on 3.9.4 right now. It builds and runs, but it doesn't format the logging properly:
```
$ gr_satellites AO-73 --wavfile satellite-recordings/ao73.wav 
gr::debug :INFO: vmcircbuf_sysconfig - Using gr::vmcircbuf_sysv_shm_factory
gr::log :DEBUG: decode_rs0 - Reed-Solomon decode corrected {:d} bytes (interleaver path {:d})
gr::log :DEBUG: decode_rs0 - Reed-Solomon decode corrected {:d} bytes (interleaver path {:d})
-> Packet from 1k2 BPSK downlink
Frame type WO10
```

It seems it's not understanding the format string and simply ignoring the rest of the parameters.

**Comment by daniestevez** [comment]
The other weird thing I found is [this compile warning](https://github.com/daniestevez/gr-satellites/runs/6012687612?check_suite_focus=true#step:4:150) about a potentially uninitialized variable, which I think didn't happen before. I checked the code and I think the compiler isn't right: `golay_res` will always be initialized at that point, because no one is changing the value of `d_use_golay` meanwhile. But I guess it doesn't hurt to initialize `golay_res` to 0 to suppress the warning.

**Comment by argilo** [comment]
> I'm testing it on 3.9.4 right now. It builds and runs, but it doesn't format the logging properly
> It seems it's not understanding the format string and simply ignoring the rest of the parameters.

Ah, that's about what I expected, although I'm surprised the extra parameter don't cause a compile error.

I guess this change will have to wait until GNU Radio 3.9 support is no longer needed (or lives on a separate branch).

**Comment by daniestevez** [comment]
Since apparently it's not so easy to get rid of boost::format in 3.9, and if I understood correctly https://github.com/gnuradio/gnuradio/pull/5645 makes gr-satellites current main not work any more with GNU Radio's main until this PR is merged, I think this maybe motivates spinning a maint-3.9 branch from main and then merging the PR to main. Any other alternatives or advice?

**Comment by argilo** [comment]
> if I understood correctly https://github.com/gnuradio/gnuradio/pull/5645 makes gr-satellites current main not work any more

It compiles, but I haven't tested whether the logging works.

**Comment by argilo** [comment]
It works fine. I guess `boost::format` is still accepted despite https://github.com/gnuradio/gnuradio/pull/5645.

**Comment by argilo** [comment]
> I think this maybe motivates spinning a maint-3.9 branch from main and then merging the PR to main. Any other alternatives or advice?

It looks like could just wait until GNU Radio 3.9 support is no longer needed. Feel free to close off this pull request if you'd like to do that.

**Comment by daniestevez** [comment]
Indeed I don't see any advantage to merging this now. But thanks so much for the effort. I'll close the PR, but when the time comes to modernize the logging, probably it's easier to start with this changeset to do most of the work.

Regarding dropping support for 3.9, I wonder if/when I'll be dropping support for 3.8, and don't see it coming soon, so probably 3.9 will be still for quite a long time.

**Comment by daniestevez** [comment]
I have needed to split the gr-satellites code bases for GR 3.9 and 3.10 due to a totally unrelated issue (`vector_type` being moved from `gr::blocks::pdu` in gr-blocks to `gr::types` in gr-runtime, which makes it too difficult to share the same code base if `vector_type` is used in the API of a C++ block).

I've recovered the change set from this PR and put it in #359. I will merge it when I've checked that CI passes.

**Pull request closed without merging**
